public/app/core/components/PageHeader/PageHeader.tsx

```diff
@@ -152,7 +166,7 @@ export default class PageHeader extends React.Component<Props, any> {
         <div className="page-container">
           <div className="page-header">
             {this.renderHeaderTitle(main)}
-            {main.children && <Navigation main={main} />}
+            {main.children && main.children.length && <Navigation main={main} />}
```

Comment:
As far as I can tell `<Navigation main={main} />` actually does not need the whole `main` `NavModelItem` here. It just needs children. As we are checking if it is not undefined we can change the props of Navigation to non optional children array which would simplify later checks a lot I think.